src/substrate/filters/enricher.ts

```diff
@@ -33,6 +34,33 @@ export async function Enrich(
       /**
        * Staking Events
        */
+      case EventKind.NewSession: {
+        const [ sessionIndex ] = event.data as unknown as [ SessionIndex ] & Codec
+        const tmp_validators = await api.query.session.validators<Vec<ValidatorId>>();
+        const currentEra = await api.query.staking.currentEra<Option<EraIndex>>();
+        let exposure : Array<Exposure>
+        let validators : Array<ValidatorId>
+        // erasStakers(EraIndex, AccountId): Exposure -> api.query.staking.erasStakers
+        if (tmp_validators && currentEra.isSome) { // if currentEra isn't empty
+          validators = tmp_validators.toArray();
+          validators.forEach(async (validator) => {
```

Comment:
Like in the other PR, since this is an async/await inside a loop, we'll need to wait for it to complete, by making it into an `await Promise.all(validators.map(async (validator) => ... ))`.